internal/services/redhatopenshift/redhatopenshift_cluster_resource.go

```diff
@@ -0,0 +1,889 @@
+	if profile == nil {
+		return []interface{}{}
+	}
+
+	podCidr := ""
+	if profile.PodCidr != nil {
+		podCidr = *profile.PodCidr
+	}
+
+	serviceCidr := ""
+	if profile.ServiceCidr != nil {
+		serviceCidr = *profile.ServiceCidr
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"pod_cidr":     podCidr,
+			"service_cidr": serviceCidr,
+		},
+	}
+}
+
+func flattenOpenShiftMasterProfile(profile *redhatopenshift.MasterProfile) []interface{} {
+	if profile == nil {
+		return []interface{}{}
+	}
+
+	subnetId := ""
+	if profile.SubnetID != nil {
+		subnetId = *profile.SubnetID
+	}
+
+	encryptionAtHostEnabled := profile.EncryptionAtHost == redhatopenshift.EncryptionAtHostEnabled
+
+	diskEncryptionSetId := ""
+
+	if profile.DiskEncryptionSetID != nil {
+		diskEncryptionSetId = *profile.DiskEncryptionSetID
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"vm_size":                    profile.VMSize,
+			"subnet_id":                  subnetId,
+			"encryption_at_host_enabled": encryptionAtHostEnabled,
+			"disk_encryption_set_id":     diskEncryptionSetId,
+		},
+	}
+}
+
+func flattenOpenShiftWorkerProfiles(profiles *[]redhatopenshift.WorkerProfile) []interface{} {
+	if profiles == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+
+	result := make(map[string]interface{})
+	result["node_count"] = int32(len(*profiles))
+
+	for _, profile := range *profiles {
+		if result["disk_size_gb"] == nil && profile.DiskSizeGB != nil {
+			result["disk_size_gb"] = profile.DiskSizeGB
+		}
+
+		if result["vm_size"] == nil && profile.VMSize != nil {
+			result["vm_size"] = profile.VMSize
+		}
+
+		if result["subnet_id"] == nil && profile.SubnetID != nil {
+			result["subnet_id"] = profile.SubnetID
+		}
+
+		if result["encryption_at_host_enabled"] == nil {
+			result["encryption_at_host_enabled"] = profile.EncryptionAtHost == redhatopenshift.EncryptionAtHostEnabled
+		}
+
+		if result["disk_encryption_set_id"] == nil && profile.DiskEncryptionSetID != nil {
+			result["disk_encryption_set_id"] = profile.DiskEncryptionSetID
+		}
+	}
+
+	results = append(results, result)
+
+	return results
+}
+
+func flattenOpenShiftAPIServerProfile(profile *redhatopenshift.APIServerProfile) []interface{} {
+	if profile == nil {
+		return []interface{}{}
+	}
+
+	return []interface{}{
+		map[string]interface{}{
+			"visibility": string(profile.Visibility),
+		},
+	}
+}
+
+func flattenOpenShiftIngressProfiles(profiles *[]redhatopenshift.IngressProfile) []interface{} {
+	if profiles == nil {
+		return []interface{}{}
+	}
+
+	results := make([]interface{}, 0)
+
+	for _, profile := range *profiles {
+		result := make(map[string]interface{})
+		result["visibility"] = string(profile.Visibility)
+
+		results = append(results, result)
+	}
+
+	return results
+}
+
+func expandOpenshiftClusterProfile(
+	input []interface{},
+	subscriptionId string,
+) *redhatopenshift.ClusterProfile {
+	resourceGroupName := fmt.Sprintf("aro-%s", randomDomainName)
```

Comment:
we should make this user configurable, maybe required, and default it to something a bit more clear:
`$clusterrgname+"-resources"` or `redhatopenshift-$name-resources`

Reply:
i suggest having it simply as a variable with no prefix and suffix, some users' needs to have resource group name in a specific fashion (if possible, with ability to add some tags)

Reply:
what i'm suggesting is that we allow people to configure this by adding it to the cluster profile block, right now we force it to be aro-abcdef which is not ideal.  So lets make it configurable and then either make it required (what i would recommend) or default to something more clear